src/Auth0Client.ts

```diff
@@ -610,6 +611,17 @@ export default class Auth0Client {
       scope: getUniqueScopes(this.defaultScope, this.scope, options.scope)
     };
 
+    return singlePromise(
+      () =>
+        this._getTokenSilently({
+          ignoreCache,
+          getTokenOptions
+        }),
+      `${this.options.client_id}::${getTokenOptions.audience}::${getTokenOptions.scope}`
+    );
```

Comment:
singlePromise is a function that is ensuring we only have a single concurrent promise based on the provided key.